### DISCUSSION TITLE
How to create path parameters and query parameters with the same name

### DISCUSSION THREAD

**Discussion by zxalif** [fastapi/fastapi#8260] [Questions] [answered]
**Description**

I was going through the [Path Parameter](https://fastapi.tiangolo.com/tutorial/path-params/) and [Query Parameters](https://fastapi.tiangolo.com/tutorial/query-params/).

So far I got, We have to create a parameter for both `Path Parameters` & `Query Parameters` in our function.

My question is what if someone wants to create an API like this. 

`https://example.com/message/{user_id}?user_id=alif&message=hello`

To create such API I have to pass two parameters with same name but which doesn't allows in Python.
```python3
@app.get('/message/{user_id}')
async def create_message(user_id: str, user_id: str, message: str):
    ......
    ......
    return {}
```
Is there any way to do this or I am missing something?

**Comment by tiangolo** [comment]
Yep. You can use an alias, and name it the same: https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#alias-parameters

**Comment by tiangolo** [comment, marked as answer]
For example:

```Python
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/test/{user_id}")
def test(user_id: str, query_user_id: str = Query(..., alias="user_id")):
    return {"user_id": user_id, "query_user_id": query_user_id}
```

It works, validates the path and query parameters, generates the OpenAPI schema, etc.

Swagger UI has a bit of trouble showing it correctly, but if you go to the URL directly: http://127.0.0.1:8000/test/foo?user_id=bar

You get the response:

```json
{"user_id":"foo","query_user_id":"bar"}
```

**Comment by zxalif** [comment]
This is awesome. 
Thank you so much!

**Comment by tiangolo** [comment]
You're welcome! And thanks for reporting back and closing the issue.

**Comment by rafaelricciweduu** [comment]
Is it possible to use alias with path parameters?

like this:
```
@router.get("/phone-numbers/{phone-number}")
def user_get_by_phone_number( phone_number: str = Path(default= None, alias="phone-number")):    
    return phone_number
```

**Comment by JarroVGIT** [comment]
Have you tried it? If so, what was the outcome? If not, please do:)

**Comment by ritesh-dubey** [comment]
@JarroVGIT Example usage of path parameters with alias
```
@app.get("/users/{id}/profile/")
def user_profile(user_id: int = Path(None, alias="id")):
    return {
        "id": user_id,
        "username": "johndoe"
    }
```
